src/CPU/ctxswitch.cpp

```diff
@@ -56,7 +56,7 @@ double total_pass() {
             exit(1);
         } else if (chid == 0) {
             char buf;
-            use_cores(vector<int> {0});
+            use_cores(vector<int> {1});
             vector<int> f(footprint_vector_len, 1);
             for (int j = 0; j < iters; j++) {
```

Comment:
What's the rationale behind doing so many context switches?
Also, whats the total number of context switches in this case? pipes or pipes -1 ?

Reply:
Same rationale as other exps, we average the value so hide the lack of precision due to clock. Total num of context switches are items * pipes, that's why we divide by that at the return statement

Reply:
Got it.
Also, why is it always the case, that for a particular value of j (inner loop), all the processes (outer loop) context switches? I get that there is only one CPU available. But, is it not possible, that one process reads from the pipe then writes, and then immediately goes to the next iteration w.o giving up the CPU to another process?

Reply:
read is a blocking operation, see https://pubs.opengroup.org/onlinepubs/007908799/xsh/read.html